Plans 2023: New 2023 plans break plans inside link-in-bio flow

claudiucelfilip opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to #
## Proposed Changes

This fixes the styling of the new Plans and Comparison grids on the `/setup/link-in-bio` flow. The issues are caused by the fact there's a different layout being used in this flow, which wasn't covered by the new Pricing Grid changes.


<img width="320" alt="216987751-2e2cc517-c31b-4ff7-a01a-934fd1c6b529" src="https://user-images.githubusercontent.com/2749938/217515868-ed9e5f8b-ccee-41ee-8362-110060418d26.png">

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Go to `/setup/link-in-bio/plans`
* The design should match the one on `/start/newsletter/plans-newsletter` on different viewport widths.

<img width="320" src="https://user-images.githubusercontent.com/2749938/217516643-437f190f-1c92-4321-9457-eaf8d7b1047f.png" />
<img width="240" src="https://user-images.githubusercontent.com/2749938/217516637-10142d07-82f4-4867-a0e0-da4123b2e7d0.png" />



## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

Addresses https://github.com/Automattic/martech/issues/1478

claudiucelfilip committed:
Remove layout constraints

claudiucelfilip committed:
Update CSS selectors to cover stepper plans

chriskmnds commented on client/my-sites/plan-features-2023-grid/style.scss:
```diff
@@ -699,18 +706,18 @@ body.is-section-signup.is-white-signup .is-onboarding-2023-pricing-grid .signup_
 			&:hover {
 				border-color: rgba(0, 0, 0, 0.04);
-				background: rgba(255, 255, 255, 0.3);
+				background: rgba(255, 255, 255, 0.3) !important;
```
As a general principle, I wish we could avoid the use of `!important`, as this cannot be overridden in any other way now.

claudiucelfilip replied:
I agree, but in this case there was no other way since there's already an `!important` defined for it [on this line](https://github.com/Automattic/wp-calypso/blob/trunk/client/landing/stepper/declarative-flow/internals/steps-repository/plans/style.scss#L158).

chriskmnds replied:
oh well 🤷 

thanks for explaining! I knew you didn't just throw it here but thought to comment anw :-)

ddc22 replied:
@chriskmnds The other day, I actually started to explore this and I even went another level up and saw another `!important`. And I decided to keep quiet 😭 .

chriskmnds reviewed (approved):
LGTM

Some observations, likely just general. Not sure if related to the changes:

- The card content/spacing looks a little more condensed in the newsletter:

| `/setup/link-in-bio/plans` | `start/newsletter/plans-newsletter` |
| ----- | ----- |
|  <img width="300" alt="Screenshot 2023-02-08 at 3 17 00 PM" src="https://user-images.githubusercontent.com/1705499/217541770-e9ca2a54-969d-46b5-b988-60553350d3f4.png"> | <img width="300" alt="Screenshot 2023-02-08 at 3 16 51 PM" src="https://user-images.githubusercontent.com/1705499/217541813-6635d352-0825-4417-a00a-da7a570dd8fd.png"> |

- Unrelated to these changes. Being sent back to the top when switching terms at the bottom feels a bit awkward:

https://user-images.githubusercontent.com/1705499/217542878-13393822-bdb7-42b2-ad70-e254c5507f29.mov

claudiucelfilip committed:
Fix tagline-price gap

claudiucelfilip committed:
Remove unnecessary CSS rule

claudiucelfilip commented:
> The card content/spacing looks a little more condensed in the newsletter:

It's somewhat related. The problem is masked on `/setup/link-in-bio` by the Enterprise tagline being on 3 lines, caused by an unexpected 1px padding in the table cell. This issue is exposed when there are only 2 lines of the tagline, in which case the prices are not being pushed down enough.
I removed the extra cell padding so the grids on both flows match, and added a min-height for the tagline so the prices don't get too close when there're fewer lines.

> Unrelated to these changes. Being sent back to the top when switching terms at the bottom feels a bit awkward

This is caused by how the interval switcher works on stepper plans. [There's a prop](https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/plans-features-main/plan-type-selector.tsx#L161) to change its behavior but removing it makes things worse (ie. page reloads on switch). 
I'd say this is outside the scope of this PR and may need further investigation.

ddc22 commented on client/my-sites/plan-features-2023-grid/style.scss:
```diff
@@ -307,6 +315,7 @@
 		border: none;
 		background-color: var(--color-surface);
 		position: relative;
+		vertical-align: baseline;
```
Hey @claudiucelfilip this results in a broken view Automattic/martech#1496
![image](https://user-images.githubusercontent.com/3422709/218315486-e2f046b7-92a3-4be7-9783-662a501eb712.png)

Can it be removed? I am trying to figure out what change it was used for. Without this line, the link-in-bio flow looks like this. Nothing seems to be broken here.


<img width="1647" alt="image" src="https://user-images.githubusercontent.com/3422709/218318359-d02a0ff3-a9c9-404d-a8da-7d0398da7db5.png">

The PR with the [change](https://github.com/Automattic/wp-calypso/pull/73157/commits/1cb4649c46b89e61c4c4f72f6f1b5c89bc39f134) above be found here -> https://github.com/Automattic/wp-calypso/pull/73157

Side Note, we need to probably remove the free plan text `Unlock a power ... ...  ... Or start with a free plan.` since we are showing the free plan here.

claudiucelfilip replied:
> Can it be removed? I am trying to figure out what change it was used for. Without this line, the link-in-bio flow looks like this. Nothing seems to be broken here.

@jdc91 Removing it causes the cells to be misaligned (on dev environment at least):

![Annotation on 2023-02-13 at 10-13-15](https://user-images.githubusercontent.com/2749938/218404838-db7b396b-baeb-4fce-b671-5c37859bf533.png)

In the other flows, the alignment it's already been set by another selector but it doesn't seem to be shared in this flow.
I think the problem is that the button cells need to be set to `bottom` and the selector for doing this (`.plan-features-2023-grid__table-item.is-top-buttons`) doesn't have enough specificity:

![Annotation on 2023-02-13 at 10-20-15](https://user-images.githubusercontent.com/2749938/218406135-565c4012-2c88-4154-aac6-99144c049c2d.png)

Adding another class seems to fix it:

![Annotation on 2023-02-13 at 10-18-56](https://user-images.githubusercontent.com/2749938/218406209-09efd521-b6ce-4d54-ba74-b5c3b6861503.png)
